src/web/client/src/components/Login/Login.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+  ComponentState,
+  Component,
+  ChangeEvent,
+  FormEvent,
+} from "react";
+import { Redirect } from "react-router-dom";
+import {
+  Container,
+  TextField,
+  Button,
+  Grid,
+  Typography,
+} from "@material-ui/core";
+
+interface Props {
+  isLoggedIn: boolean;
+  history: any;
+  onAuthChange: (authState: boolean) => void;
+}
+interface State {
+  username: string;
+  password: string;
+}
+
+/**
+ * A Login form for users to login 
+ */
+export default class Login extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      username: "",
+      password: "",
+    };
+  }
+
+  /**
+   * Sets the appropriate state when a keyboard event is triggered
+   * @params event - An event that records changes happening to an input field
+   */
+  handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
+    event.preventDefault();
+    const target: HTMLInputElement = event.target as HTMLInputElement;
+    const label: string = target.name;
+    const value: string = target.value;
+    this.setState({ [label]: value } as ComponentState);
+  };
+
+  /**
+   * Submits the user credentials to the auth server
+   * @params event - An event that records submission of a form element
+   *
+   * Changes the root level isLoggedIn state to True and redirects to the
+   * "/streams" route if login submission is successful.
+   * Otherwise, alert the user that login is not successful.
+   */
+  loginSubmit = async (event: FormEvent<HTMLFormElement>) => {
```

Comment:
yes like i said, you could've called loginSubmit differently to get away from that browser-dependent code

Reply:
yes